src/app/utils/firebase.js

```diff
@@ -16,171 +16,502 @@ const FireBaseTools = {
+          // Add the JSON date key
+          switch (startDate.getDay()) {
+
+            case 0: // Sunday
+              if (givenWeekDay[0] === 'Y') {
+                timetable[newDateObject] = { description: 'No Description' };
+              }
+              break;
+
+            case 1: // Monday
+              if (givenWeekDay[1] === 'Y') {
+                timetable[newDateObject] = { description: 'No Description' };
+              }
+              break;
+
+            case 2: // Tuesday
+              if (givenWeekDay[2] === 'Y') {
+                timetable[newDateObject] = { description: 'No Description' };
+              }
+              break;
+
+            case 3: // Wednesday
+              if (givenWeekDay[3] === 'Y') {
+                timetable[newDateObject] = { description: 'No Description' };
+              }
+              break;
+
+            case 4: // Thursday
+              if (givenWeekDay[4] === 'Y') {
+                timetable[newDateObject] = { description: 'No Description' };
+              }
+              break;
+
+            case 5: // Friday
+              if (givenWeekDay[5] === 'Y') {
+                timetable[newDateObject] = { description: 'No Description' };
+              }
+              break;
+
+            case 6: // Saturday
+              if (givenWeekDay[6] === 'Y') {
+                timetable[newDateObject] = { description: 'No Description' };
+              }
+              break;
+
+            default:
+              console.log('There was an error in retrieving the day of the week');
+            //TODO add a proper error check?
+          }
+        }
+
+        // check next date (startDate acts as our iterator in this loop so it takes the value of the next day)
+        const newDate = startDate.setDate(startDate.getDate() + 1);
+        startDate = new Date(newDate);
+      }
+      // return a promise of a timetable object with all the dates of a given course section path
+      return timetable;
+    }).catch(error => ({
+      errorCode: error.code,
+      errorMessage: error.message,
+    }));
+    /* eslint-enable */
+  },
+
+  getSections: (courseName) => {
+    const sections = [];
+    /* eslint-disable */
+    return sectionsRef.child(courseName).once('value').then(function(snap) {
+      snap.forEach(function(childSnap) {
+        sections.push({section:childSnap.key, maxPat:childSnap.child('MaxPat').val(), component:childSnap.child('Component').val()})
+      });
+      return sections;
+    }).catch(error => ({
+      errorCode: error.code,
+      errorMessage: error.message,
+    }));
+    /* eslint-enable */
+  },
+
+   getUserEvents: () => {
+    let userCourses = null;
+    const stringCourses = [];
+    const id = firebaseAuth.currentUser ? firebaseAuth.currentUser.uid : null;
+    /* eslint-disable */
+    return usersRef.child(id.toString()).once('value').then(function(snap) {
```

Comment:
Using the .once function means that we aren't listening for changes, only retrieving the data one time. I think possibly in sprint 5, if we give the TA the capability to change individual class descriptions, then we will need to ensure that students retrieve the changes in real time. In order to do that, we will need to use the .on() function instead. Of course, this involves more complicated logic, so we wouldn't want to do it until we really need it. Just something to remember for later on.